packages/pages/src/dev/server/ssr/findMatchingStaticTemplate.ts

```diff
@@ -0,0 +1,20 @@
+import { ViteDevServer } from "vite";
+import { isStaticTemplateConfig } from "../../../common/src/feature/features.js";
+import { findTemplateModuleInternal } from "./findTemplateModuleInternal.js";
+
+export default async function findMatchingStaticTemplate(
+  vite: ViteDevServer,
+  slug: string,
+  templateFilepaths: string[]
+) {
+  return findTemplateModuleInternal(
+    vite,
+    (t) => {
+      if (!isStaticTemplateConfig(t.config)) {
+        return false;
+      }
+      return slug === t.getPath({});
```

Comment:
I wonder if this could be an issue. Obviously for static files we want a static url, but I wonder if it's possible to pass in props in the static template and that would mess things up. Hopefully that's handled or caught earlier and a user can't do that. Probably not an issue with your change here though.

Reply:
I was thinking for static pages it might be nice to have the module export a static string instead of a function like
```
export const path = 'static-url.html';
```
but I'm not sure I like having multiple ways to specify the path so unsure for now

Reply:
Originally I was also worried about somebody's getPath erroring out, but my thought was that if you have a static page and do anything except directly return a string, then you're not following the contract of static templates.